text/0013-abort-pending-updates.md

```diff
@@ -0,0 +1,191 @@
+- Title: Cancel pending tasks
+- Start Date:
+- Specification PR: https://github.com/meilisearch/specifications/pull/13
+- MeiliSearch Issue: 
+ - https://github.com/meilisearch/MeiliSearch/issues/1104
+- Parent Spec: https://github.com/meilisearch/specifications/pull/12
+
+
+# Cancel pending tasks
+
+## 1. Feature Description and Interaction
+
+### Summary
+
+Allow to cancel tasks in the task queue or to cancel the entire coming tasks in the queue. A canceled task will have the status `canceled`. Only pending tasks will be cancellable. 
+
+### Motivation
+
+The task queue is convenient because it allows us to serialize tasks (updates/config/dumps) and assure the [ACID property](https://en.wikipedia.org/wiki/ACID) of MeiliSearch. It also causes some problems. As the task queue is an append-only queue, it is impossible to remove a task from this queue. The tasks can be large and can take a long time to index. Sometimes, it would be convenient to cancel a message in this queue or all the next pending tasks on it. 
+
+### Additional Materials
+
+As explained in the [parent spec](https://github.com/meilisearch/specifications/pull/12), It's quite rare to have this kind of task queue. Elasticsearch or Algolia, who have this kind of task queue, don't give the cancel or clear their 'tasks' queues. 
+
+### Explanation
+
+Two new routes and a new status (`canceled`) must be added to make this feature possible. 
+
+A canceled task will stay at its place in the queue. So the typical answer of the get all task status route will be: 
+
+```json
+[
+ {
+ "status": "processed",
+ "taskId": 1,
+ ...
+ },
+ {
+ "status": "failed",
+ "taskId": 2,
+ ...
+ },
+ {
+ "status": "processed",
+ "taskId": 3,
+ ...
+ },
+ {
+ "status": "pending",
+ "taskId": 4,
+ ...
+ },
+ {
+ "status": "canceled",
+ "taskId": 5,
+ ...
+ },
+ {
+ "status": "pending",
+ "taskId": 6,
+ ...
+ },
+ ...
+]
+```
+
+the task status document will look like:
+
+```json
+{
+ "status": "canceled",
+ "taskId": 12,
+ "type": {
+ "name": "DocumentsAddition",
+ "number": 4
+ },
+ "duration": 0,
+ "enqueuedAt": "2019-12-07T21:16:09.623944Z",
+ "processedAt": null,
+ "canceledAt": "2019-12-07T21:16:09.703509Z", // date of cancelation
+}
+```
+
+Until now, we will consider that every task status documents returned will have all fields. Even empty ones. 
```

Comment:
I don't undesratnd this sentence, could you rephrase it, please?

Reply:
@Kerollmops I didn't wrote this specification! I paused it when I arrived to take a new look at it after the 0.21 release :) Probably soon since I'm preparing the next releases cycle 😀